src/lib/tls/tls12/msg_client_kex.cpp

```diff
@@ -133,6 +133,15 @@ Client_Key_Exchange::Client_Key_Exchange(Handshake_IO& io,
                                                           rng,
                                                           policy);
 
+         // RFC 8422 - 5.11.
+         //   With X25519 and X448, a receiving party MUST check whether the
+         //   computed premaster secret is the all-zero value and abort the
+         //   handshake if so, as described in Section 6 of [RFC7748].
```

Comment:
I suppose it doesn't hurt to do that check in TLS 1.3 as well. It should go into `tls_extensions_key_share.cpp` line 109 after calling `Callbacks::tls_ephemeral_key_agreement`.